### PR TITLE
Replace Rtags with LSP + Clangd

### PR DESCRIPTION
**Pull request by gleon99**

### CODE REVIEW

File: init.el [hunk 1]
```diff
@@ -462,40 +462,13 @@
...
-
-  (use-package company-rtags
-    :ensure
-    :after company
-    :config (push 'company-rtags company-backends))
-
-  (use-package flycheck-rtags
-    :ensure
-    :after rtags))
+      (defvar company-backends)))
```
Comment (drrlvn) [inline init.el]: What is this for?

Comment (gleon99) [reply init.el]: I didn't add this line - see line 473 in the old version.

Comment (drrlvn) [reply init.el]: But you deleted a bunch of lines and I think this is no longer needed.

Comment (gleon99) [reply init.el]: You're right, it's not needed anymore.

**Pull request merged**